### ISSUE TITLE
Firebase Console returns broken image

### ISSUE THREAD

**Issue by codeashian** [benawad/apollo-server-react-file-upload#17]
Hi!

First, thanks for your tutorial on Youtube! It was great. Though, I can't seem to fix one problem when using `createReadStream` and the bucket storage.

When uploading the  image to firebase, the image is broken and I'm not able to preview it in the Firebase Console. All I can see is an error saying *Error loading preview*. But the filesize, mimetype and accessToken are being uploaded, so i'm doing something right.

Here's the part of my Resolver function where i'm uploading an image:

*PS. The bucket comes from an Apollo datasource, that's why I can access it using `this`.*

```
 const { createReadStream, filename, mimetype } = image;

  try {
    await new Promise((res, rej) =>
      createReadStream()
        .pipe(
          this.bucket.file(filename).createWriteStream({
            resumable: false,
            gzip: true,
            metadata: {
              contentType: mimetype,
              metadata: {
                firebaseStorageDownloadTokens: uuidv4(),
              },
            },
          })
        )
        .on("finish", res)
        .on("error", rej)
    );
    return filename;
  } catch (err) {
    return new Error(err)
  }
```

Do you have any suggestions? 

I would really appreciate your help. I've been googling for 2 days now :(

**Comment by benawad** [comment]
nothing looks amiss from your code snippet, I'm using almost the exact same code to upload to google cloud storage.

have you tried visiting the url instead of viewing it through firebase's preview?

**Comment by codeashian** [comment]
Yes, and the URL is broken too. I also tried to just save it to my filesystem, but that's broken too. It seems like the image content is not being processed. 

I'm not sure if it's a server or client issue! 

I'm using `apollo-server-lambda` as server, have you used that before?

**Comment by benawad** [comment]
I haven't, but I'm not sure if lambda gives you access to a filesystem you can save the image to
